spec/services/record_declarations/ecf/early_career_teacher_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe RecordDeclarations::ECF::EarlyCareerTeacher do
+  let(:cpd_lead_provider) { create(:cpd_lead_provider) }
+  let(:another_lead_provider) { create(:cpd_lead_provider, name: "Unknown") }
+  let(:ect_profile) { create(:early_career_teacher_profile) }
+  let(:mentor_profile) { create(:mentor_profile) }
+  let(:induction_coordinator_profile) { create(:induction_coordinator_profile) }
+  let(:params) do
+    {
+      raw_event: "{\"participant_id\":\"37b300a8-4e99-49f1-ae16-0235672b6708\",\"declaration_type\":\"started\",\"declaration_date\":\"2021-06-21T08:57:31Z\",\"course_identifier\":\"ecf-induction\"}",
+      participant_id: ect_profile.user_id,
+      declaration_date: "2021-06-21T08:46:29Z",
+      declaration_type: "started",
+      course_identifier: "ecf-induction",
+      cpd_lead_provider: another_lead_provider,
+    }
+  end
+  let(:ect_params) do
+    params.merge({ cpd_lead_provider: cpd_lead_provider })
+  end
+  let(:mentor_params) do
+    ect_params.merge({ participant_id: mentor_profile.user_id, course_identifier: "ecf-mentor" })
+  end
+  let(:induction_coordinator_params) do
+    ect_params.merge({ participant_id: induction_coordinator_profile.user_id })
+  end
+  let(:delivery_partner) { create(:delivery_partner) }
+  let!(:school_cohort) { create(:school_cohort, school: ect_profile.school, cohort: ect_profile.cohort) }
+  let!(:partnership) do
+    create(:partnership,
+           school: ect_profile.school,
+           lead_provider: cpd_lead_provider.lead_provider,
+           cohort: ect_profile.cohort,
+           delivery_partner: delivery_partner)
+  end
```

Comment:
This setup seems very similar between tests. Can we see if there is some way of commonising it?

Reply:
Probably not in this PR.